src/CosmosDB/CosmosDB/SQL/NewAzCosmosDbClientEncryptionKey.cs

```diff
@@ -0,0 +1,160 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.CosmosDB.Models;
+using Microsoft.Azure.Commands.CosmosDB.Exceptions;
+using Microsoft.Azure.Management.CosmosDB;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Azure.Core;
+using Azure.Core.Cryptography;
+using Microsoft.Rest.Azure;
+using System.Security.Cryptography;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using Microsoft.Azure.PowerShell.Cmdlets.CosmosDB.Exceptions;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDbClientEncryptionKey", DefaultParameterSetName = NameParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSSqlClientEncryptionKeyGetResults), typeof(ConflictingResourceException))]
```

Comment:
```suggestion
    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDbClientEncryptionKey", DefaultParameterSetName = NameParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSSqlClientEncryptionKeyGetResults), typeof(ConflictingResourceException))]
```
To support SupportsShouldProcess, please add shouldprocess logic in your command, here is an example: https://github.com/Azure/azure-powershell/blob/2cf0575bb404f1ae3cb85f675f07a0fbfee47af6/src/KeyVault/KeyVault/Commands/Key/AddAzureKeyVaultKey.cs#L323

Reply:
yes its done on line 164.